python/aitubertalk/rooms/room_module.py

```
@@ -0,0 +1,357 @@
+            
+            self._client.logger.info("Successfully left room")
+            
+        except AITuberTalkError:
+            raise
+        except Exception as e:
+            self._client.logger.error(f"Room leave failed: {e}")
+            raise AITuberTalkError(
+                code=ErrorCode.UNKNOWN_ERROR,
+                message=f"Failed to leave room: {e}",
+                retryable=True
+            )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)